sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py

```diff
@@ -2,11 +2,176 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-from typing import TYPE_CHECKING
+from typing import cast, Optional, Union, TYPE_CHECKING
+
+from cryptography.hazmat.primitives.asymmetric.padding import AsymmetricPadding, OAEP, PKCS1v15, PSS, MGF1
+from cryptography.hazmat.primitives.asymmetric.rsa import (
+    RSAPrivateKey,
+    RSAPrivateNumbers,
+    RSAPublicKey,
+    RSAPublicNumbers,
+)
+from cryptography.hazmat.primitives.asymmetric.utils import Prehashed
+from cryptography.hazmat.primitives.hashes import Hash, HashAlgorithm, SHA1, SHA256, SHA384, SHA512
+from cryptography.hazmat.primitives.serialization import Encoding, KeySerializationEncryption, PrivateFormat
+
+from ._enums import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
 
 if TYPE_CHECKING:
-    from . import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
-    from typing import Any, Optional
+    # Import clients only during TYPE_CHECKING to avoid circular dependency
+    from ._client import CryptographyClient
+    from .._client import KeyClient
+
+
+SIGN_ALGORITHM_MAP = {
+    SHA256: SignatureAlgorithm.rs256,
+    SHA384: SignatureAlgorithm.rs384,
+    SHA512: SignatureAlgorithm.rs512,
+}
+OAEP_MAP = {
+    SHA1: EncryptionAlgorithm.rsa_oaep,
+    SHA256: EncryptionAlgorithm.rsa_oaep_256
+}
+PSS_MAP = {
+    SignatureAlgorithm.rs256: SignatureAlgorithm.ps256,
+    SignatureAlgorithm.rs384: SignatureAlgorithm.ps384,
+    SignatureAlgorithm.rs512: SignatureAlgorithm.ps512,
+}
+
+
+class ManagedRsaKey(RSAPrivateKey):
+    """An `RSAPrivateKey` implementation based on a key managed by Key Vault.
```

Comment:
Why not attempt to download the public key and do public key operations like encrypt or verify as well? .NET and Go do this. It would make for a more complete implementation.

Basically,
```mermaid
flowchart LR
  style S1 fill:green
  style S2 fill:green
  style E1 fill:red
  style E2 fill:red
  S1([URI or KeyClient])-->A{Public key operation?}--Yes-->B{Can download key?}--Yes-->C{Has private key material?}--Yes-->E1(Local operation)
  S2([JWK])-->C
  B--No-->E2(Remote operation)
  C--No-->E2
  A--No-->C
```

Reply:
Resolving; discussed this offline and we'll be tackling public key operations in a `KeyVaultRSAPublicKey` type that implements the `RSAPublicKey` interface.